p2pclient/p2pclient.go

```diff
@@ -0,0 +1,125 @@
+package p2pclient
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"sync"
+
+	ggio "github.com/gogo/protobuf/io"
+	logging "github.com/ipfs/go-log"
+	pb "github.com/libp2p/go-libp2p-daemon/pb"
+	multiaddr "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr-net"
+)
+
+var log = logging.Logger("p2pclient")
+
+// MessageSizeMax is cribbed from github.com/libp2p/go-libp2p-net
+const MessageSizeMax = 1 << 22 // 4 MB
+
+// Client is the struct that manages a connection to a libp2p daemon.
+type Client struct {
+	control manet.Conn
+
+	mhandlers sync.Mutex
+	handlers  map[string]io.Closer
+}
+
+// NewClient creates a new libp2p daemon client, connecting to a daemon
+// listening at the provided multiaddr.Multiaddr
+func NewClient(addr multiaddr.Multiaddr) (*Client, error) {
+	conn, err := manet.Dial(addr)
```

Comment:
Can't vouch for it, but I don't think there's a manet dailer for unix sockets. Should we validate that the arg is of type "unix"?

Reply:
We can also not use multiaddrs at all.

Reply:
Agree. We do need some flexibility to add SHM in the future.

Reply:
Let's not overdesign for that yet though.